scripts/west_commands/run_common.py

```diff
@@ -224,35 +235,46 @@ def do_run_common(command, user_args, user_runner_args, domain_file=None):
             # single-use commands in a dictionary so that they get executed
             # once per unique board name.
             if cache['BOARD_DIR'] not in processed_boards and 'SOC_FULL_DIR' in cache:
-                soc_yaml_file = Path(cache['SOC_FULL_DIR']) / 'soc.yml'
-                board_yaml_file = Path(cache['BOARD_DIR']) / 'board.yml'
-                group_type = 'boards'
+                highest_priority = IGNORED_RUN_ONCE_PRIORITY
+                check_files = []
+                # TODO: need a list of the paths to search provided by the build system here
+                check_files.append(SocBoardFilesProcessing(Path(cache['SOC_FULL_DIR']) / 'soc.yml'))
+                check_files.append(SocBoardFilesProcessing(Path(cache['BOARD_DIR']) / 'board.yml', True))
 
-                # Search for flash runner configuration, board takes priority over SoC
-                try:
-                    with open(board_yaml_file, 'r') as f:
-                        data_yaml = yaml.safe_load(f.read())
+                for check in check_files:
+                    try:
+                        with open(check.filename, 'r') as f:
+                            check.yaml = yaml.safe_load(f.read())
 
-                except FileNotFoundError:
-                    continue
+                            if 'runners' not in check.yaml:
+                                continue
+                            elif check.board is False and 'run_once' not in check.yaml['runners']:
+                                continue
 
-                if 'runners' not in data_yaml:
-                    # Check SoC file
-                    group_type = 'qualifiers'
-                    try:
-                        with open(soc_yaml_file, 'r') as f:
-                            data_yaml = yaml.safe_load(f.read())
+                            if 'priority' in check.yaml['runners']:
+                                check;priority = check.yaml['runners']['priority']
+                            else:
+                                check.priority = BOARD_FILE_RUN_ONCE_DEFAULT_PRIORITY if check.board is True else SOC_FILE_RUN_ONCE_DEFAULT_PRIORITY
+
+                            if check.priority == highest_priority:
+                                log.wrn("Duplicate flash run once configuration found, this will not work")
```

Comment:
If this will not work, then this should not be a warning.
```suggestion
                                log.die('no two flash runner configurations can have equal priority')
```

Reply:
Fixed